src/mscorlib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -60,32 +60,33 @@
 **    may be delayed until appdomain shutdown.
 ===========================================================*/
 
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Threading;
+
 namespace System.Runtime.CompilerServices
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Runtime.Versioning;
-    using System.Runtime.InteropServices;
-
-
     #region ConditionalWeakTable
-    [System.Runtime.InteropServices.ComVisible(false)]
+    [ComVisible(false)]
     public sealed class ConditionalWeakTable<TKey, TValue>
         where TKey : class
         where TValue : class
     {
+        #region Fields
+        private const int InitialCapacity = 8;  // Initial length of the table. Must be a power of two.
+        private readonly object _lock;          // This lock protects all mutation of data in the table.  Readers do not take this lock.
+        private volatile Container _container;  // The actual storage for the table; swapped out as the table grows.
+        #endregion
 
-        #region Constructors
-        [System.Security.SecuritySafeCritical]
+        #region Constructors / Finalizers
         public ConditionalWeakTable()
         {
-            _buckets = Array.Empty<int>();
-            _entries = Array.Empty<Entry>();
-            _freeList = -1;
-            _lock = new Object();
-
-            Resize();   // Resize at once (so won't need "if initialized" checks all over)
+            _lock = new object();
+            _container = new Container(this);
+            GC.SuppressFinalize(this);
         }
+
+        ~ConditionalWeakTable() { } // to satisfy contract
```

Comment:
@weshaggard, can we remove the finalizer from the contract? Can we remove all finalizes on sealed types like this from the contracts?